### DISCUSSION TITLE
Confused whether to go with web3.py or web3.js

### DISCUSSION THREAD

**Discussion by abdullahpichoo** [smartcontractkit/full-blockchain-solidity-course-py#1645] [Q&A] [answered]
Hi! I started a few months ago and I've completed Lesson 6. I just recently saw that Patrick uploaded a new 32-hour-long course about blockchain dev but that one is based on javascript. 
I was confused about whether to keep on learning the python one or should I start the Javascript one. Which would be better for the future?

The question might sound a bit childish but I would really appreciate some advice from someone experienced.

Thanks

**Comment by cromewar** [comment, marked as answer]
Hello @abdullahpichoo 
Well I would said it does not matter that much, the main language for smart contracts is Solidity and it's the core of learning web3 development for Ethereum. The framework you choose to deploy and test your smart-contracts is up to you, at the end you will end up learning more than one and as you are starting I would said just choose the one you feel more confortable with.
If you like Python keep with it, if you like JS go ahead, there is not a something better over the other.
Cheers.

**Reply by abdullahpichoo** [reply]
Thank you so much for the clarification!
